util/llbutil/authprovider/cloudauth/gcp_login.go

```diff
@@ -0,0 +1,47 @@
+package cloudauth
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/docker/cli/cli/config/types"
+	"golang.org/x/oauth2/google"
+)
+
+const (
+	OauthScope     = "https://www.googleapis.com/auth/cloud-platform"
+	OauthTokenUser = "oauth2accesstoken"
+)
+
+func (ap *authProvider) getAuthConfigGCR(ctx context.Context, fullPathPrefix, pathPrefix, org, project, host string) (*authConfig, error) {
+	jsonKeyPath := pathPrefix + "GCP_KEY"
+	ap.console.VerbosePrintf("looking up GCP_KEY", jsonKeyPath)
+	keyJSON, err := ap.getProjectOrUserSecret(ctx, org, project, jsonKeyPath)
+	if err != nil {
+		return nil, err
+	}
+	keyJSON = strings.TrimSpace(keyJSON)
+	if keyJSON == "" {
+		return nil, fmt.Errorf("%sGCP_KEY is missing (or empty), but %scred_helper was set to gcp-login", fullPathPrefix, fullPathPrefix)
+	}
+
+	ap.console.VerbosePrintf("creating new a new JWT config for %s", host)
+
+	jwtCFG, err := google.JWTConfigFromJSON([]byte(keyJSON), OauthScope)
```

Comment:
missing err handling?

Reply:
good catch, looks like I removed it by mistake.
That makes me wonder though - should I add integration tests to for the err path (secret is missing, json in the secret is invalid, etc)?